loading.bs

```diff
@@ -0,0 +1,965 @@
+1. If |certResponse|'s [=response/body=] is null or that body's [=body/stream=]
+    is null, return a failure.
+1. Let |bytes| be the result of [=ReadableStream/read all bytes|reading all
+    bytes=] from |certResponse|'s [=response/body=]'s [=body/stream=] with a
+    [=ReadableStream/get a reader|new reader=] over the same stream.
+1. Wait for |bytes| to settle.
+1. If |bytes| was rejected, return a failure.
+1. Let |chain| be the [=certificate chain=] produced by parsing |bytes|' value
+    using the [=cert-chain CDDL=]. If |bytes|'s value doesn't match this CDDL or
+    isn't [=canonically-encoded CBOR=], return a failure.
+1. Assert: |chain| has at least one [=list/item=].
+1. If the [=SHA-256=] hash of |chain|'s [=certificate chain/leaf=]'s [=augmented
+    certificate/certificate=] is not equal to |certSha256|, return a failure.
+1. Return |chain|.
+
+<h3 algorithm id="signed-message">The signed message</h3>
+
+The <dfn>b1 signed message</dfn> for an [=exchange signature=] |signature| and a
+[=byte sequence=] |headerBytes| is the concatenation of the following [=byte
+sequences=]:
+
+1. The byte 0x20 (SP) repeated 64 times. This matches the TLS 1.3 ([[RFC8446]])
+    format to avoid cross- protocol attacks if anyone uses the same key in a TLS
+    certificate and an exchange-signing certificate.
+1. `` `HTTP Exchange 1 b1` `` to serve as a context string.
+
+    Note: Each draft of
+    [[draft-yasskin-httpbis-origin-signed-exchanges-impl-01]] and the final RFC
+    for [[draft-yasskin-http-origin-signed-responses]] will use distinct context
+    strings.
+1. A single 0x00 byte which serves as a separator.
+1. The concatenation of the following [=byte sequences=], which is the
+    [=canonically-encoded CBOR=] for a CBOR map:
+    1. 0xA5, the CBOR header for a 5-item map.
+    1. 0x64 followed by `` `date` ``
+    1. The [=CBOR integer encoding=] of |signature|'s [=exchange
+        signature/date=].
+    1. 0x67 followed by `` `expires` ``.
+    1. The [=CBOR integer encoding=] of |signature|'s [=exchange
+        signature/expiration time=].
+    1. 0x67 followed by `` `headers` ``.
+    1. |headerBytes|.
+    1. 0x6B followed by `` `cert-sha256` ``.
+    1. The [=CBOR byte string encoding=] of |signature|'s [=exchange
+        signature/certSha256=].
+    1. 0x6C followed by `` `validity-url` ``.
+    1. The [=CBOR byte string encoding=] of |signature|'s [=exchange
+        signature/validityUrlBytes=].
+
+<h3 algorithm id="validating-signature">Validating a signature</h3>
+
+An [=exchange signature=] |signature| <dfn for="exchange signature" lt="is
+valid|is not valid">is valid</dfn> for a [=byte sequence=] |headerBytes| if the
+following steps return `valid`:
+
+1. Let |clockSkew| be the uncertainty in the UA's estimate of the current time
+    caused by clock skew on the client. The UA MAY set this to 0 or use a more
+    sophisticated estimate.
+1. If the UA's estimate of the current time is more than |clockSkew| before
```

Comment:
@davidben @sleevi How many hives does this bit give you?

Reply:
I'm not going to have bandwidth to review this this week, in part because I need to go get treated for hives. It quite concerns me, and I can tell it's going to take time and focus that I don't yet have to adequately review :/